src/GraphQL/Types/Composite/ComplexGraphType.cs

```diff
@@ -17,6 +17,9 @@ public abstract class ComplexGraphType<TSourceType> : GraphType, IComplexGraphTy
         /// <inheritdoc/>
         protected ComplexGraphType()
         {
+            if (typeof(IGraphType).IsAssignableFrom(typeof(TSourceType)) && GetType() != typeof(Introspection.__Type))
```

Comment:
🤔 Maybe in future we introduce some flag (GlobalSwitch?) to allow this behavior.

Reply:
I understand that for 99,99% cases this check helps to solve developer problems.